### PR TITLE
chore: rename darwin to macos

### LINKED ISSUE #10118: chore(feedback): use `MacOS` instead of `Darwin` for system name
### Is your enhancement related to a problem? Please describe

As required in https://github.com/podman-desktop/podman-desktop/pull/10116#issuecomment-2500727481 we should use `MacOS` instead of `Darwin` for system name on darwin/macos platform.

### Describe the solution you'd like

Replace in the sys-info utils the darwin hardcoded value by macos.

https://github.com/podman-desktop/podman-desktop/blob/fdc22edac9b6cc2b2e342f8506f1a7134ab878ba/packages/main/src/plugin/util/sys-info.ts#L72

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Rename `Darwin` to `MacOS` (more user friendly)

### What issues does this PR fix or reference?

Fixes https://github.com/podman-desktop/podman-desktop/issues/10118

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
Issue: it needs to be macOS not MacOS

File: packages/main/src/plugin/util/sys-info.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ class DarwinInfo extends SystemInfo {
    * so we use release to get the os version (E.g. 24.1.0), and arch (E.g. arm64)
    */
   override getSystemName(): string {
-    return `Darwin ${release()} - ${arch()}`;
+    return `MacOS ${release()} - ${arch()}`;
```
Comment (axel7083) [inline packages/main/src/plugin/util/sys-info.ts]:
```suggestion
    return `macOS ${release()} - ${arch()}`;
```

File: packages/main/src/plugin/util/sys-info.spec.ts [hunk 2]
```diff
@@ -92,5 +92,5 @@ test('darwin should use os#release and os#arch', () => {
   expect(version).not.toHaveBeenCalled();
   expect(arch).toHaveBeenCalled();
 
-  expect(result).toBe('Darwin 23.9.4 - arm64');
+  expect(result).toBe('MacOS 23.9.4 - arm64');
```
Comment (axel7083) [inline packages/main/src/plugin/util/sys-info.spec.ts]:
```suggestion
  expect(result).toBe('macOS 23.9.4 - arm64');
```

**Review by benoitf — APPROVED** [review]
Thx

**Pull request merged**